jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -324,23 +326,62 @@ def plot_selected_points(self):
                                                            getattr(y, 'value', y))
 
     def vue_zoom_in(self, *args, **kwargs):
-        """This function will zoom into the image based on the selected points"""
+
+        self.zoom_to_selected()
+
+    def zoom_to_selected(self, padding=0.02, return_bounding_box=False):
+        """
+        Zoom on the default viewer to a region containing the currently selected
+        points in the catalog.
+
+        Parameters
+        ----------
+        padding : float, optional
+            A fractional value representing the padding around the bounding box
+            of the selected points. It is applied as a proportion of the largest
+            dimension of the current extent of loaded data. Defaults to 0.02.
+        return_bounding_box : bool, optional
+            If True, returns the bounding box of the zoomed region as
+            ((x_min, x_max), (y_min, y_max)). Defaults to False.
+
+        Returns
+        -------
+        list of float or None
+            If there are activley selected rows, and ``return_bounding_box`` is
+            True, returns a list containing the bounding
+            box coordinates: [x_min, x_max, y_min, y_max].
+            Otherwise, returns None.
+
+        """
+
+        viewer = self.app._jdaviz_helper._default_viewer
+
         selected_rows = self.table.selected_rows
 
+        if not len(selected_rows):
+            return
+
+        if padding <= 0 or padding > 1:
+            raise ValueError("`padding` must be between 0 and 1.")
+
         x = [float(coord['x_coord']) for coord in selected_rows]
         y = [float(coord['y_coord']) for coord in selected_rows]
 
+        limits = viewer.state._get_reset_limits()
```

Comment:
What if the viewer and the image have different dimensions? 

I was testing the default Imviz notebook and one case I noticed was the viewer limits were dependent on my browser instance and how I defined the width but the image keeps the same aspect ratio in the viewer when I resize the browser.

Reply:
I'm honestly not sure what the answer to this is

Reply:
You can reproduce where this case causes and issue with the zoom with the following:
```
catalog = imviz.plugins['Catalog Search']._obj
catalog.open_in_tray()
catalog.catalog.selected = 'Gaia'
catalog.search()

catalog.table.selected_rows = catalog.table.items[:2]
catalog.zoom_to_selected()
```

Reply:
Is the concern just about reproducibility?  Or is it somehow failing to zoom to include all selected sources?

Reply:
Can we get the limits from the selected_rows instead of the viewer?

Reply:
viewer.state._get_reset_limits() is always the same no matter how the browser is resized, correct?

Reply:
At a high level, user wants zoom w.r.t. image dimension. Here is how astrowidgets defined it:

https://github.com/spacetelescope/jdaviz/blob/5cd23a8ec6813274938e45dcdc3110c40c1c728f/jdaviz/core/astrowidgets_api.py#L175-L178

However, there is a known issue where its implementation is currently broken if images are linked by WCS because of the fake WCS layer (e.g., a very small image in pixel dimension) is used as reference. Maybe it won't be hard to fix if we grab the top visible data layer and then do a `pixel_to_pixel` conversion. Just that the ticket is never high enough priority to pull up into sprint because astrowidgets is low priority for Indigo right now.

Not sure how much this helps...

Reply:
> Is the concern just about reproducibility? Or is it somehow failing to zoom to include all selected sources?

It is failing to include all selected sources

Reply:
I couldn't reproduce a case where resizing the browser excluded any of the selected points

Reply:
https://github.com/user-attachments/assets/218ace46-142c-4052-b6a1-087ff4ee40bc

Reply:
can you see what is being used as padding?  And the output of `viewer.state._get_reset_limits()` (maybe this is a bug there)?  My other idea is that this is just an issue with providing x/y limits as opposed to center and radius, since the rectangle of limits we pass is not exactly what ends up adopted.

Reply:
I'd still say this is an improvement over the previous behavior and maybe we get it in as is so that its in for the demos and make a tech-debt ticket to make more general in the future?  If we're worried, we can hide `padding` from the user API until then.

Reply:
ok i was able to reproduce the issue, very weird.

Reply:
I would say this is still an improvement, because the same zoom scenario on main in the example doesn't work at all and the image is totally zoomed out. This is at least in the right area of the image now when it fails after resizing your browser window. We should look into what is happening though.

Reply:
For the browser width in the video:
**limits**:
(-0.5, 4733.5, -0.5, 4718.5)
**max_dim**:
4734.0
**padding**:
94.68
**before applying padding (x_min, x_max) (y_min, y_max)**:
2202.56701 2589.79442
2243.31321 2716.02305
**after applying padding (x_min, x_max) (y_min, y_max)**:
2107.8870100000004 2684.47442
2148.63321 2810.7030499999996

**then calling viewer.get_limits()**:
(2156.4823647015282, 2635.8790652984726, 2365.1348456839964, 2594.2014143160036)

Reply:
**limits:**
(-0.5, 4733.5, -0.5, 4718.5)
**max_dim:**
4734.0
**padding:**
94.68
**before applying padding (x_min, x_max) (y_min, y_max)**
2202.56701 2589.79442
2243.31321 2716.02305
**after applying padding (x_min, x_max) (y_min, y_max)**
2107.8870100000004 2684.47442
2148.63321 2810.7030499999996


After Zooming:

**viewer.get_limits()**
(2156.4823647015282, 2635.8790652984726, 2365.1348456839964, 2594.2014143160036)

**viewer.state._get_reset_limits()**
(-0.5, 4733.5, -0.5, 4718.5)

Reply:
I am going to approve the PR now so we can get this into 4.1 this morning and have access to the logic for AAS. @cshanahan1 and I investigated a bit more on Friday and saw an issue arise after the call to `set_limits`, I will make a follow up ticket to continue investigating, including testing the increasing to padding and see if it is enough to resolve the issue or if the limits are being reset somewhere else that we will need to address.

Reply:
https://github.com/spacetelescope/jdaviz/pull/3371